cdc_stream/main.cc

```diff
@@ -0,0 +1,155 @@
+ABSL_FLAG(
+    uint16_t, ssh_port, 22,
+    "Port to use while connecting to the remote instance being streamed to.");
+ABSL_FLAG(std::string, ssh_command, "",
+          "Path and arguments of ssh command to use, e.g. "
+          "C:\\path\\to\\ssh.exe -F config_file. Can also be specified by the "
+          "CDC_SSH_COMMAND environment variable.");
+ABSL_FLAG(std::string, scp_command, "",
+          "Path and arguments of scp command to use, e.g. "
+          "C:\\path\\to\\scp.exe -F config_file. Can also be specified by the "
+          "CDC_SCP_COMMAND environment variable.");
+ABSL_FLAG(int, verbosity, 2, "Verbosity of the log output");
+
+namespace {
+
+constexpr char kHelpText[] = R"!(Stream files from a Windows to a Linux device
+
+Usage: cdc_stream [flags] start windows_dir [user@]host:linux_dir
+       Streams the Windows directory windows_dir to directory linux_dir on the
+       Linux host using SSH username user.
+ 
+       cdc_stream [flags] stop host
+       Stops the streaming session to the given Linux host.)!";
+
+// Splits |user_host_dir| = [user@]host:dir up into [user@]host and dir.
+// Does not touch Windows drives, e.g. C:\foo.
+bool ParseUserHostDir(const std::string& user_host_dir, std::string* user_host,
+                      std::string* dir) {
+  std::vector<std::string> parts =
+      absl::StrSplit(user_host_dir, absl::MaxSplits(':', 1));
+  if (parts.size() < 2 ||
+      (parts[0].size() == 1 && toupper(parts[0][0]) >= 'A' &&
+       toupper(parts[0][0]) <= 'Z')) {
+    LOG_ERROR(
+        "Failed to parse '%s'. Make sure it is of the form "
+        "[user@]host:linux_dir.",
+        user_host_dir);
+    return false;
+  }
+
+  *user_host = parts[0];
+  *dir = parts[1];
+  return true;
+}
+
+std::string GetFlagFromEnvOrArgs(const char* env,
+                                 const absl::Flag<std::string>& flag) {
+  std::string value = absl::GetFlag(flag);
+  if (value.empty()) {
+    cdc_ft::path::GetEnv(env, &value).IgnoreError();
+  }
+  return value;
+}
+
+}  // namespace
+
+int main(int argc, char* argv[]) {
+  absl::SetProgramUsageMessage(kHelpText);
+  std::vector<char*> args = absl::ParseCommandLine(argc, argv);
+
+  uint16_t service_port = absl::GetFlag(FLAGS_service_port);
+  int verbosity = absl::GetFlag(FLAGS_verbosity);
+  std::string ssh_command =
+      GetFlagFromEnvOrArgs("CDC_SSH_COMMAND", FLAGS_ssh_command);
+  std::string scp_command =
+      GetFlagFromEnvOrArgs("CDC_SCP_COMMAND", FLAGS_scp_command);
+  uint16_t ssh_port = absl::GetFlag(FLAGS_ssh_port);
+
+  cdc_ft::LogLevel level = cdc_ft::Log::VerbosityToLogLevel(verbosity);
+  cdc_ft::Log::Initialize(std::make_unique<cdc_ft::ConsoleLog>(level));
+
+  if (args.size() < 2) {
+    LOG_INFO(kHelpText);
+    return 0;
+  }
+  std::string command = args[1];
+  if (command != "start" && command != "stop") {
+    LOG_ERROR("Unknown command '%s'. Must be 'start' or 'stop'.", command);
```

Comment:
All of `cdc_stream`, `cdc_stream --help` and `cdc_stream --helpfull` print this message. I've added a note about --helpfull since that shows the flags.